### PR TITLE
Fix issue 20589 - typeof may give wrong result in case of classes defining `opCall` operator

### PR DESCRIPTION
**Pull request by drug007**
See bugzilla [here](https://issues.dlang.org/show_bug.cgi?id=20589)

### CODE REVIEW

**Comment by drug007** [comment]
@wilzbach could you point out where can I add a test for this issue?

**Comment by thewilsonator** [comment]
Please change the commit message to say `Fix issue 20589`.

**Comment by Geod24** [comment]
> Please change the commit message to say `Fix issue 20589`.

`Fix issue 20589 - <Issue title>`

People watching the core repositories don't want to open every single notification they get, just those that seems relevant to their area of expertise. Likewise, when going through old PR / commits (trying to find where a bug was fixed, regression introduced, or feature added), it is extremely painful to see commit without correct title / description.

**Comment by Geod24** [comment]
Thanks for changing the title, but the bot picks up the commit, so you also have to go and edit it.

File: std/container/array.d [hunk 1]
```diff
@@ -1585,6 +1585,19 @@ if (!is(Unqual!T == bool))
     ai.insertBack(arr);
 }
 
+@system unittest
```
Comment (atilaneves) [inline std/container/array.d]: Is there a reason this can't be `@safe`?

Comment (drug007) [reply std/container/array.d]: destructor std.container.array.Array!(MyClass).Array.~this is @system

Comment (atilaneves) [reply std/container/array.d]: Could you please add a comment explaining that then? Thanks!

Comment (drug007) [reply std/container/array.d]: Done.

File: std/container/array.d [hunk 2]
```diff
@@ -1585,6 +1585,21 @@ if (!is(Unqual!T == bool))
     ai.insertBack(arr);
 }
 
+// destructor std.container.array.Array!(MyClass).Array.~this is @system
+// so the unittest is @system too
```
Comment (atilaneves) [inline std/container/array.d]: Sorry to have missed this the last time around, I think this just needs a reference (with a link) to the issue on bugzilla and this should be good to go.

Comment (drug007) [reply std/container/array.d]: Sorry that I didn't do it myself. I promise the next time I'll be a good boy!

**Pull request merged**